src/dao/DeviceDAO.js

```diff
@@ -18,6 +25,18 @@ class DeviceDAO extends DAO<Device, DeviceMutator> {
       translator: new DeviceTranslator(),
     });
   }
+
+  fetchParticleAttributes(deviceID: EntityID): LoadObject<ParticleAttributes> {
+    const funcString = 'Default.particleAttributes()';
+    const stringifiedID = deviceID.toString();
+
+    const handler = this.__buildHandler({}, false).find(
+      this.__reformatIDValue(stringifiedID),
+    );
+    handler.func(funcString);
+
+    return this.__fetchCustom(handler, {}, `${funcString}${deviceID}`);
```

Comment:
Doesn't this already include the device ID from the handler in the cache key?

Reply:
not, only queryOptions and this string from the argument.

Reply:
Ahh.. I thought deviceID would be part of queryOptions.